iceoryx_hoofs/test/moduletests/test_posix_posix_call.cpp

```diff
@@ -812,5 +918,9 @@ TEST_F(PosixCall_test, ErrnoIsSetFromReturnValueWhenFunctionHandlesErrnosInRetur
             EXPECT_THAT(r.errnum, Eq(RETURN_VALUE));
         });
 
-    EXPECT_FALSE(internal::GetCapturedStderr().empty());
+    if (iox::testing::TestingLogger::doesLoggerSupportLogLevel(iox::log::LogLevel::ERROR))
```

Comment:
Just to prevent a misunderstanding. Do you suggest to remove the if-clause when moved into a function? This is required when log messages are expected. The idea behind the if-clause is to prevent failed checks when the minimal log level compiled into the binary is higher than the level of the expected log messages.

Having all of this in a function would reduce code duplication though.

Reply:
@ithier I'm dispassionate about this. I see the point @FerdinandSpitzschnueffler though since it adds 5 lines for each check. What do you think about something like this

```cpp
void checkLogMessages(std::function<void(const std::vector<std::string>&)> check)
{
    if (iox::testing::TestingLogger::doesLoggerSupportLogLevel(iox::log::LogLevel::ERROR))
    {
        check(iox::testing::TestingLogger::getLogMessages());
    }
}

TEST(Foo, Bar)
{
    ...
    checkLogMessages([](auto logMessages) { ASSERT_EQ(logMessages.size(), 0U); });
}
```

While writing this, the idea emerged to have the `checkLogMessages` function as part of  the testing logger, e.g.
```cpp
void checkLogMessageIfLogLevelIsSupported(iox::log::LogLevel logLevel,
                                          std::function<void(const std::vector<std::string>&)> check)
{
    if (doesLoggerSupportLogLevel(iox::log::LogLevel::ERROR))
    {
        check(getLogMessages());
    }
}
```
It would then be used like
```cpp
checkLogMessageIfLogLevelIsSupported(iox::log::LogLevel::ERROR,
                                     [](auto logMessages) { ASSERT_EQ(logMessages.size(), 0U); });
```

Then `TestingLogger::getLogMessages()` could be made private. It is currently only used in one place where the pattern from `checkLogMessageIfLogLevelIsSupported` is not strictly followed and that place could easily be refactored.

What do you think?

Reply:
@elBoberido Works for me. I will update the `TestingLogger`.

cc. @FerdinandSpitzschnueffler

Reply:
@elBoberido Done: https://github.com/eclipse-iceoryx/iceoryx/pull/1787/files#diff-c7f77bd3c5a06b57d4f4d925461f943204707b6e0156002f96f452e9e3e2c6b3R87

cc. @FerdinandSpitzschnueffler